RecoBTag/ONNXRuntime/plugins/DeepBoostedJetTagsONNXProducer.cc

```diff
@@ -0,0 +1,277 @@
+
+  FloatArrays data_;
+
+  bool debug_ = false;
+};
+
+DeepBoostedJetTagsONNXProducer::DeepBoostedJetTagsONNXProducer(const edm::ParameterSet &iConfig,
+                                                               const ONNXRuntime *cache)
+    : src_(consumes<TagInfoCollection>(iConfig.getParameter<edm::InputTag>("src"))),
+      flav_names_(iConfig.getParameter<std::vector<std::string>>("flav_names")),
+      debug_(iConfig.getUntrackedParameter<bool>("debugMode", false)) {
+  // load preprocessing info
+  const auto &prep_pset = iConfig.getParameterSet("preprocessParams");
+  input_names_ = prep_pset.getParameter<std::vector<std::string>>("input_names");
+  for (const auto &group_name : input_names_) {
+    const auto &group_pset = prep_pset.getParameterSet(group_name);
+    input_shapes_.push_back(group_pset.getParameter<std::vector<unsigned>>("input_shape"));
+    auto &prep_params = prep_info_map_[group_name];
+    prep_params.var_length = group_pset.getParameter<unsigned>("var_length");
+    prep_params.var_names = group_pset.getParameter<std::vector<std::string>>("var_names");
+    const auto &var_info_pset = group_pset.getParameterSet("var_infos");
+    for (const auto &var_name : prep_params.var_names) {
+      const auto &var_pset = var_info_pset.getParameterSet(var_name);
+      double median = var_pset.getParameter<double>("median");
+      double norm_factor = var_pset.getParameter<double>("norm_factor");
+      prep_params.var_info_map[var_name] = PreprocessParams::VarInfo(median, norm_factor);
+    }
+
+    // create data storage with a fixed size vector initilized w/ 0
+    unsigned len = prep_params.var_length * prep_params.var_names.size();
+    data_.emplace_back(len, 0);
+  }
+
+  if (debug_) {
+    for (unsigned i = 0; i < input_names_.size(); ++i) {
+      const auto &group_name = input_names_.at(i);
+      std::cout << group_name << "\nshapes: ";
+      for (const auto &x : input_shapes_.at(i)) {
+        std::cout << x << ", ";
+      }
+      std::cout << "\nvariables: ";
+      for (const auto &x : prep_info_map_.at(group_name).var_names) {
+        std::cout << x << ", ";
+      }
+      std::cout << "\n";
+    }
+  }
+
+  // get output names from flav_names
+  for (const auto &flav_name : flav_names_) {
+    produces<JetTagCollection>(flav_name);
+  }
+}
+
+DeepBoostedJetTagsONNXProducer::~DeepBoostedJetTagsONNXProducer() {}
+
+void DeepBoostedJetTagsONNXProducer::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  // pfDeepBoostedJetTags
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("src", edm::InputTag("pfDeepBoostedJetTagInfos"));
+  edm::ParameterSetDescription preprocessParams;
+  preprocessParams.setAllowAnything();
+  desc.add<edm::ParameterSetDescription>("preprocessParams", preprocessParams);
+  desc.add<edm::FileInPath>("model_path",
+                            edm::FileInPath("RecoBTag/Combined/data/DeepBoostedJet/V01/full/resnet.onnx"));
```

Comment:
it looks like the path is incorrect or inconsistent with https://github.com/cms-data/RecoBTag-Combined/pull/25 which has this file in V02

Reply:
My bad -- in the local test area I have converted model for both V01 and V02, but I did not add V01 to the cms-data PR as I think we don't need them. I fixed the path here to V02 in https://github.com/cms-sw/cmssw/pull/28112/commits/d96e9a03fd39f27a2b7a545bc102e1caba15ea3d.